versioned_docs/version-2.x/magic-button/contribution-mode.mdx

```diff
@@ -0,0 +1,130 @@
+    // add-next-line
+    "./node_modules/front-commerce/modules/magic-button",
+  ],
+  serverModules: [
+    { name: "FrontCommerce", path: "server/modules/front-commerce" },
+    { name: "Magento2", path: "server/modules/magento2" },
+  ],
+  webModules: [
+    { name: "FrontCommerce", path: "./src/web" },
+    { name: "ThemeChocolatine", path: "./theme-chocolatine/web" },
+  ],
+};
+```
+
+The **Magic Button** is only visible when the **Contribution Mode** is active.
+
+## How to activate Contribution Mode
+
+### Activate with Config Provider
+
+The default activation for **Contribution Mode** is defined by a the
+[contributionModeConfigProvider](https://gitlab.blackswift.cloud/front-commerce/front-commerce/-/blob/e79a04bf0fd4c171b1530e67f7ac1b53cabed83a/src/server/modules/Front-Commerce/Contribution-Mode/contribution-mode/contributionModeConfigProvider.js).
+
+You can follow the
+[Override an existing configuration](/docs/2.x/advanced/server/configurations#override-an-existing-configuration)
+to update the value.
+
+:::caution
+
+While in development you can use the
+`FRONT_COMMERCE_CONTRIBUTION_MODE_FORCE=true` environment variable to force the
+contribution mode to be enabled, but this should not be used in production.
+
+:::
+
+### Activate with Loader
+
+To activate the **Contribution Mode** through a loader, you can take advantage
+of the `forceEnablePreviewMode` option in the
+[ContributionModeLoader](https://gitlab.blackswift.cloud/front-commerce/front-commerce/-/blob/7a2aee7e11c7bd45cfc3a5a940407f9e220e66f4/src/server/modules/Front-Commerce/Contribution-Mode/contribution-mode/loaders/ContributionModeLoader.js#L57-60)
+
+```js
+export default {
+    namespace:"MyModule"
+    dependencies:["Front-Commerce/Contribution-Mode"]
+    resolvers:{
+        Mutation:{
+            myQuery: async (root, args, context, info) => {
+                context.loaders.ContributionMode.forceEnablePreviewMode()
+                return true
+            }
+        }
+    }
+}
+```
+
+This will activate the Contribution Mode and the Preview Mode.
+
+## Check if Contribution Mode is active
+
+You can check if the **Contribution Mode** is active by using the `enabled`
+value in the ContributionModeLoader
+
+```js
+export default {
+    namespace:"MyModule"
+    dependencies:["Front-Commerce/Contribution-Mode"]
+    resolvers:{
+        Query:{
+            myQuery: async (root, args, context, info) => {
+                return context.loaders.ContributionMode.enabled
+            }
+        }
+    }
+}
+```
+
+Or alternatively you can use the
+[Feature Flags](/docs/2.x/advanced/features/feature-flags) the check if it has
+been enabled
+
+```js
+export default {
+    namespace:"MyModule"
+    dependencies:["Front-Commerce/Contribution-Mode"]
+    resolvers:{
+        Query:{
+            myQuery: async (root, args, context, info) => {
+                return context.loaders.FeatureFlag.isActive("contributionMode")
+            }
+        }
+    }
+}
```

Comment:
**nitpick:** I'm not in favor of documenting several ways to do the same thing. We should enforce one only.

**suggestion:** either the feature flag or the contribution mode loader must be documented.

**question:** What's the most recommended API?